core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,475 @@
+          fileOutputStream.write(noteText.toByteArray())
+          fileOutputStream.close()
+          context.toast(R.string.note_save_successful, Toast.LENGTH_SHORT)
+          noteEdited = false // As no unsaved changes remain
+          enableDeleteNoteMenuItem()
+        } catch (e: IOException) {
+          e.printStackTrace()
+            .also { context.toast(R.string.note_save_unsuccessful, Toast.LENGTH_LONG) }
+        }
+      } else {
+        context.toast(R.string.note_save_successful, Toast.LENGTH_LONG)
+        Log.d(TAG, "Required folder doesn't exist")
+      }
+    } else {
+      context.toast(R.string.note_save_error_storage_not_writable, Toast.LENGTH_LONG)
+    }
+  }
+
+  private fun deleteNote() {
+    val notesFolder = File(zimNotesDirectory)
+    val noteFile =
+      File(notesFolder.absolutePath, "$articleNoteFileName.txt")
+    val noteDeleted = noteFile.delete()
+    if (noteDeleted) {
+      add_note_edit_text.text.clear()
+      disableMenuItems()
+      context.toast(R.string.note_delete_successful, Toast.LENGTH_LONG)
+    } else {
+      context.toast(R.string.note_delete_unsuccessful, Toast.LENGTH_LONG)
+    }
+  }
+
+  /* String content of the note text file given at:
+   *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+   * is displayed in the EditText field (note content area)
+   */
+  private fun displayNote() {
+
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    if (noteFile.exists()) {
+      writeNote(noteFile)
+    }
+
+    // No action in case the note file for the currently open article doesn't exist
+  }
+
+  private fun writeNote(noteFile: File) {
+    noteFileExists = true
+    val contents = StringBuilder()
+    try {
+      noteFile.bufferedReader().use { bufferReader ->
+        var line: String?
+        while (bufferReader.readLine().also { line = it } != null) {
+          contents.append(line)
+          contents.append(System.getProperty("line.separator"))
+        }
+      }
+    } catch (e: IOException) {
+      e.printStackTrace()
+      Log.d(TAG, "Error reading line with BufferedReader")
+    }
+    add_note_edit_text.setText("$contents") // Display the note content
+    add_note_edit_text.setSelection(add_note_edit_text.text.length - 1)
+    enableShareNoteMenuItem() // As note content exists which can be shared
+    enableDeleteNoteMenuItem()
+  }
+
+  private fun shareNote() {
+
+    /* The note text file corresponding to the currently open article, given at:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * is shared via an app-chooser intent
+     * */
+    if (noteEdited) {
+      // Save edited note before sharing the text file
+      saveNote(add_note_edit_text.text.toString())
+    }
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    var noteFileUri: Uri? = null
+    if (noteFile.exists()) {
+      noteFileUri = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
noteFileUri should probably have its scope reduced to only inside this `if` and its declaration joined with this assignment.
The intent would naturally move in here too